anvil/src/shell/mod.rs

```diff
@@ -313,12 +313,21 @@ fn ensure_initial_configure(surface: &WlSurface, space: &Space<WindowElement>, p
     };
 }
 
-fn place_new_window(space: &mut Space<WindowElement>, window: &WindowElement, activate: bool) {
-    // place the window at a random location on the primary output
+fn place_new_window(
+    space: &mut Space<WindowElement>,
+    pointer_location: Point<f64, Logical>,
+    window: &WindowElement,
+    activate: bool,
+) {
+    // place the window at a random location on same output as pointer
     // or if there is not output in a [0;800]x[0;800] square
     use rand::distributions::{Distribution, Uniform};
 
-    let output = space.outputs().next().cloned();
+    let output = space
+        .output_under(pointer_location)
+        .next()
```

Comment:
I hadn't realized `output_under` returned an Iterator... I wouldn't have mentioned the change had I known. Still LGTM though.